apps/omg_watcher_rpc/lib/web/controllers/transaction.ex

```diff
@@ -90,14 +89,9 @@ defmodule OMG.WatcherRPC.Web.Controller.Transaction do
 
   defp increment_metrics_counter(response) do
     case response do
-      {:error, {:validation_error, _, _}} ->
-        Metrics.increment("transaction.failed.validation", 1)
-
-      %Plug.Conn{} ->
-        Metrics.increment("transaction.succeed", 1)
-
-      _ ->
-        Metrics.increment("transaction.failed.unidentified", 1)
+      {:error, {:validation_error, _, _}} -> :telemetry.execute([:transaction], %{counter: 1}, %{valid: :failed})
```

Comment:
similar to the above comments and potterhat's approach, these clauses should be function headers in the telemetry handler

Reply:
similar answer, telemetry handler shouldn't calculate anything.